src/ocl/convolutionocl.cpp

```diff
@@ -3365,7 +3365,7 @@ void ConvolutionDescriptor::ConvolutionBackwardWeights(Handle& handle,
         decltype(auto) direction      = conv::Direction::BackwardWeights;
         decltype(auto) algorithm_name = AlgorithmName{ConvolutionAlgoToDirectionalString(
             static_cast<miopenConvAlgorithm_t>(algo), direction)};
-        decltype(auto) ctx = conv::ProblemDescription{xDesc, dwDesc, dyDesc, *this, direction};
+        decltype(auto) ctx = conv::ProblemDescription{dyDesc, dwDesc, xDesc, *this, direction};
```

Comment:
@DrizztDoUrden AFAICS this is the only place where order of tensors in the initializer is NOT `{input, weights, output}`. This looks suspicious, I need more explanation than commit description ("Fixed missing invoker error").

Reply:
Check miopen::ProblemDescription constructors. It swaps in with out on backward directions.

Reply:
- (1) So we are going to stop swapping in the new descriptor?
- (2) Why don't we use old miopen::ProblemDescription here?